### ISSUE TITLE
[Feature] MSX light gun emulation [sf#115]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#139]
**Reported by manuelbi on 2005-03-27 12:28 UTC**
Similar to what RedMSX 1.4 has: emulation of the light
gun. Like in Karoshi's Duck Hunt and later maybe the
light gun of ASCII (The Plus-X Terminator Laser).
This could be done by using the cursor as a pointing
device again (so, make a nice cursor for this, not hide
it :)

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2007-05-25 17:45 UTC**
Logged In: YES 
user_id=78178
Originator: YES

Here's some info from NYYRIKKI, posted here: http://www.msx.org/forumtopicl6873.html

The correct distance to use the gun seems to be about 80cm between 14"
monitor (CM8833, older model) and head of the gun. At this distance
the circle, that is visible to the transistor seems to have only about
5 pixel radius after that the light effect to result will drop
locarithmically very fast. The transistor will activate pretty quickly
after detecting the light. Delay is normally only about one line (*).

When the capasitor is fully loaded it will hold the signal up about 31
lines. How well the capasitor is loaded depends of the brighness of
the pixels inside the circle. This capasitor gets loaded to almost
full pretty fast so I would say, that normal hit will cause signal to
go on for 24-28 lines. The gun is more sensitive to green and blue
light than red light.

Note:
Measurements have been made with "normal" screen brighness in "normal"
home environment (40W light bulp on roof) with "normal" screen height
settings. Changeing these parameters will change the results. Propably
manufacture date of the light gun will also have effect.

\* All lines I talk about are NTSC scanlines (with incomplete timing as
normal in MSX)

**Comment by openMSX-import** [comment]
**Commented by eadmaster on 2014-07-12 08:19 UTC**
I second the request since there are some games that can only be played with the gunstick:
Guillermo Tell
Solo
Mike Gunner
Space Smugglers
Target Plus
Trigger

Also, there are some with only support the ASCII Plus-X Terminator Laser, like Dungeon Hunter.

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2014-07-12 09:58 UTC**
What exactly is gunstick?

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2014-07-12 13:03 UTC**
OK, apparently it's the same as light gun :)

See also http://www.msxblue.com/manual/controlsother_c.htm

**Comment by Tovarichtch** [comment]
I've wanted to create a feature request, but found this issue already present there. 7 years later, is there any plan to integrate Gunstick / Terminator Laser in openMSX ? There are great games using the light gun and I would love to make them work. Actually, blueMSX has this support since long time ago but it's broken : trigger work, but not aiming even if I see my cursor on screen.

Thanks in advance !

**Comment by MBilderbeek** [comment]
How many games are there? Only a handful, right?

**Comment by Tovarichtch** [comment]
From what I've collected, here's the list : 

1. Air Hockey
2. Dungeon Hunter
3. Shooting Collection (which had 5 short games : 4 shooting type, 1 race ("Pig Race"))
4. Duck Hunt (with the Gunstick patch)

And all the Gunstick version games

1. Bestial Warrior (the character move itself, you shoot ennemies)
2. Cosmic Sheriff
3. Guillermo Tell
4. Gunstreet
5. Mike Gunner
6. Solo
7. Sootland
8. Space Smugglers
9. Target Plus
10. The A Team
11. Trigger

And probably some I'm missing ([bigger list here](https://computeremuzone.com/?id=games&cat=8&val=Gunstick)).

**Comment by MBilderbeek** [comment]
OK, and how do you expect it to be emulated? Some kind of cross-hair that can be controlled with the mouse?

**Comment by Tovarichtch** [comment]
Yes, as expected with blueMSX (basic crosshair controlled by the mouse input) but not working sadly (on their latest build from 2009). But the crosshair is not mandatory. Playing with modern light guns such as the Sinden Lightgun is more than enough to aim.

**Comment by Darknior** [comment]
Excellent idea, i stand for this feature too :)
Thanks
